### PR TITLE
Feature 3511 add pagination on new products page

### PR DESCRIPTION
**Pull request by dborovsky**

### CODE REVIEW

**Comment by dborovsky** [comment]
Please note,  I didn`t add filter page on New Products Page, because you didn`t answer me.  But I added all properties for administrator to set everything for filter pagination and of course pagination pages itself.

**Comment by dborovsky** [comment]
@skoshelev  any news?

**Comment by skoshelev** [comment]
Hi @dborovsky Please make all your changes in one commit. And also note that you created new settings but did not specify default values, also added a new model but did not add locales and did not display settings in the UI.

**Comment by dborovsky** [comment]
@skoshelev ok? all clear. I will fix it

**Comment by dborovsky** [comment]
@skoshelev Setting catalog has already field "Number of products on 'New products' page"

if I set here for example 3, "New products" page will display 3 items per page. It working. 
Must I add new setting field "Number of products per page on 'New products' page"?

Field "Number of products on 'New products' page" must set total new items?

**Comment by skoshelev** [comment]
@dborovsky Yes, I think that you need to add the ability to specify how many products should appear on the one page

**Comment by dborovsky** [comment]
@skoshelev I undestood this. My question was:  Do we need  specify total displaying new products or only specify how many products should appear on the one page?

**Comment by skoshelev** [comment]
Hi @dborovsky. I think we need both settings, please leave the setting for total number of products

**Comment by dborovsky** [comment]
@skoshelev I merged with branch develop. When I save setting get error: 
**SqlException: Too many arguments were specified for the ProductTagCountLoadAll procedure or function.**

**Comment by dborovsky** [comment]
@skoshelev any news

**Comment by skoshelev** [comment]
Hi @dborovsky. I'm sorry, we decided not to include this task in 4.20. I haven’t seen your changes yet, but don’t worry, I’ll check everything when the opportunity arises. Thank you for your help

File: src/Presentation/Nop.Web/Factories/ICatalogModelFactory.cs [hunk 1]
```diff
@@ -208,5 +208,15 @@ public partial interface ICatalogModelFactory
...
         #endregion
+
+        #region New products
+        
+        /// <summary>
+        /// Prepare products for New products page
+        /// </summary>
+        /// <param name="command">Catalog paging filtering command</param>
+        /// <returns>Products by tag model</returns>
+        NewProductsModel PrepaireNewProductsModel(CatalogPagingFilteringModel command);
```
Comment (poseidon87) [inline src/Presentation/Nop.Web/Factories/ICatalogModelFactory.cs:219]: Prepare is misspelled within this method name ;)

**Comment by RomanovM** [comment]
@dborovsky Thanks, we used your contribution to implement this feature.

**Pull request closed without merging**